### PR TITLE
Fix measure

### PR DESCRIPTION
**Pull request by GiggleLiu**
list of changes
1. `measure!(op, reg, loc)`
direct measure of operators.
2. `insert_qubit!`
insert qubits into register
3. fix `measure(reg, loc)` behavior when reg is focused.

### CODE REVIEW

File: src/Blocks/block_operations.jl [hunk 1]
```diff
@@ -103,10 +103,29 @@ function expect end
...
 
-expect(op::MatrixBlock, dm::DensityMatrix) = mapslices(x->sum(mat(op).*x)[], dm.state, dims=[1,2]) |> vec
+#expect(op::MatrixBlock, dm::DensityMatrix) = mapslices(x->sum(mat(op).*x)[], dm.state, dims=[1,2]) |> vec
 expect(op::MatrixBlock, dm::DensityMatrix{1}) = sum(mat(op).*dropdims(dm.state, dims=3))
+function expect(op::AbstractBlock, dm::DensityMatrix{B}) where B
+    mop = mat(op)
+    [tr(view(dm.state,:,:,i)*mop) for i=1:B]
+end
+
+expect(op::AbstractBlock, reg::AbstractRegister{1}) = reg'*apply!(copy(reg), op)
```
Comment (Roger-luo) [inline src/Blocks/block_operations.jl]: What's `reg'` is it `ConjReg` ? I think we should rename this to `AdjointReg` which is more consistent to stdlib.

File: docs/src/man/blocks.md [hunk 2]
```diff
@@ -16,13 +16,6 @@ a quantum circuit into serveral kinds of blocks. The uppermost abstract type for
 
 ## Composite Blocks
 
-### Roller
```
Comment (Roger-luo) [inline docs/src/man/blocks.md]: Why roller is deleted here? We haven't move them outside yet.

File: src/Blocks/block_operations.jl (hunk 1 above)
Comment (GiggleLiu) [reply src/Blocks/block_operations.jl]: Yes, it is. +1 for this change.

File: docs/src/man/blocks.md (hunk 2 above)
Comment (GiggleLiu) [reply docs/src/man/blocks.md]: This is documentation, we should introduce most basic & important blocks first.

File: src/Blocks/CachedBlock.jl [hunk 3]
```diff
@@ -39,7 +39,8 @@ clear!(c::CachedBlock) = (clear!(c.server, c.block); c)
 # forward methods
 function mat(c::CachedBlock)
     if !iscached(c.server, c.block)
-        m = dropzeros!(mat(c.block))
+        #m = dropzeros!(mat(c.block); trim=1e-16)
+        m = mat(c.block)
```
Comment (Roger-luo) [inline src/Blocks/CachedBlock.jl]: Why `dropzeros!` is removed? Is this causing any bug? @GiggleLiu

**Comment by Roger-luo** [comment]
@GiggleLiu I think this PR and ConjReg are in YaoArrayRegister now, double check this and we could close this PR.

I'm not sure what was the bug about measure, so I don't if this is fixed in YaoArrayRegister or not. Double check?

**Comment by Roger-luo** [comment]
This is done in other packages.

**Pull request closed without merging**